### PR TITLE
add documentation for workload mappings

### PR DESCRIPTION
**Pull request by sadlerap**
I'd like more examples, but I think this is a good start.

### CODE REVIEW

File: application-operator.md [hunk 1]
```diff
@@ -143,7 +143,81 @@ TODO
...
+   ("foo['bar']") operators.  For more information, please refer to [the
+   spec](https://github.com/servicebinding/spec#fixed-jsonpath).
+
+   This field indicates where annotations for the generated pod might be found.
+   Defaults to `.spec.template.metadata.annotations`.
+5. Entries in this array define where containers might be found.  This lets a
+   single version refer to multiple container fields, such as containers and
+   init containers.
+6. Defines the root of where container information may be found.
+7. Indicates the name of the container.  Defaults to `.name`.
```
Comment (scothis) [inline application-operator.md:197]: The spec does not define a default value for name, unlike env and volumeMounts.

```suggestion
7. Indicates the name of the container.  If not specified, selective binding to containers by name is not supported for this mapping.
```

https://github.com/servicebinding/runtime/blob/main/apis/v1beta1/clusterworkloadresourcemapping_webhook.go#L133-L136

File: application-operator.md [hunk 2]
```diff
@@ -143,7 +143,81 @@ TODO
...
+8. Indicates where environment variables can be set in the container.  Defaults to `.env`.
+9. Indicates where volume mount information can be set in the container.  Defaults to `.volumeMounts`.  
+10. Indicates where volumes can be specified for the generated pod.  Corresponds to the volumes field in pods.
```
Comment (scothis) [inline application-operator.md:200]:
```suggestion
10. Indicates where volumes can be specified for the generated pod.  Defaults to `.spec.template.spec.volumes`.
```

**Review by scothis — COMMENTED** [review]
Looks good. Mostly polishing suggestions, although there is one technical correction.

File: application-operator.md (hunk 1 above)
Comment (sadlerap) [reply application-operator.md]: Good catch!  Fixed in c2993ed.

File: application-operator.md (hunk 2 above)
Comment (sadlerap) [reply application-operator.md]: Added in c2993ed.

File: application-operator.md [hunk 3]
```diff
@@ -143,7 +143,81 @@ TODO
...
+4. Some fields in the mapping must conform to what the spec calls a "Fixed
+   JSONPath".  This is a JSONPath, but restricted to dot (".") and array access
+   ("foo['bar']") operators.  For more information, please refer to [the
+   spec](https://github.com/servicebinding/spec#fixed-jsonpath).
+
+   This field indicates where annotations for the generated pod might be found.
+   Defaults to `.spec.template.metadata.annotations`.
+5. Entries in this array define where containers might be found.  This lets a
+   single version refer to multiple container fields, such as containers and
+   init containers.
```
[the remark this answers is not in the document]
Comment (sadlerap) [reply application-operator.md]: Added in c2993ed.

File: application-operator.md [hunk 4]
```diff
@@ -143,7 +143,81 @@ TODO
...
+
+Some notes:
+
+1. These resources need to be named after the plural name of the resource.  For
+   instance, to define mappings for `CronJob` resources, the mapping needs to
+   be named `cronjobs.batch`.
+2. Entries in this array define the resource version that this template maps
+   to.  This lets one mapping support multiple resource versions.
+3. The version here must either match a version exactly (e.g. `v1alpha1`) or
+   can be `*` to match all versions that don't have an exact match.
```
[the remark this answers is not in the document]
Comment (sadlerap) [reply application-operator.md]: Added in c2993ed.

File: application-operator.md [hunk 5]
```diff
@@ -143,7 +143,81 @@ TODO
...
+      env:          # string (Fixed JSONPath), optional (8)
+      volumeMounts: # string (Fixed JSONPath), optional (9)
+    volumes:        # string (Fixed JSONPath), optional (10)
+```
+
+Some notes:
+
+1. These resources need to be named after the plural name of the resource.  For
+   instance, to define mappings for `CronJob` resources, the mapping needs to
+   be named `cronjobs.batch`.
```
[the remark this answers is not in the document]
Comment (sadlerap) [reply application-operator.md]: Added in c2993ed.

File: application-operator.md [hunk 6]
```diff
@@ -143,7 +143,81 @@ TODO
...
+2. Entries in this array define the resource version that this template maps
+   to.  This lets one mapping support multiple resource versions.
```
[the remark this answers is not in the document]
Comment (sadlerap) [reply application-operator.md]: Added in c2993ed.

File: application-operator.md [hunk 7]
```diff
@@ -143,7 +143,81 @@ TODO
...
-TODO
+By default, an implementation assumes that it is projecting binding information
+into workloads with a shape similar to `Deployment.apps` resources.  For
+instance, it may assume that it can find volume information at
+`.spec.template.spec.volumes`.
+
+However, this assumption does not work with workloads of a different resource
+shape, such as a CronJob.  To support these cases, you can use
+`ClusterWorkloadResourceMapping.servicebinding.io` resources to inform the
+ServiceBinding controller how it can project information into these resources.
```
[the remark this answers is not in the document]
Comment (sadlerap) [reply application-operator.md]: Added in c2993ed.

**Review by scothis — APPROVED** [review]

**Pull request merged**